Source/WebCore/PAL/pal/cf/CoreAudioExtras.h

```diff
@@ -53,4 +55,27 @@ inline std::span<const AudioBuffer> span(const AudioBufferList& list)
     return unsafeMakeSpan(list.mBuffers, list.mNumberBuffers);
 }
 
-} // namespace WebCore
+inline size_t allocationSize(const AudioBufferList& list)
+{
+    return offsetof(AudioBufferList, mBuffers) + sizeof(AudioBuffer) * std::max<uint32_t>(1U, list.mNumberBuffers);
+}
+
+// AudioBufferList is a variable-length struct, so create on the heap with a generic new() operator
+// with a custom size, and initialize the struct manually.
+enum class ShouldZeroMemory : bool { No, Yes };
+inline std::unique_ptr<AudioBufferList, WTF::SystemFree<AudioBufferList>> createAudioBufferList(uint32_t bufferCount, ShouldZeroMemory shouldZeroMemory)
+{
+    CheckedSize bufferListSize = offsetof(AudioBufferList, mBuffers);
+    bufferListSize += CheckedSize { sizeof(AudioBuffer) } * std::max<uint32_t>(1, bufferCount);
```

Comment:
What’s the goal of the checking here? Crash on overflow?

Reply:
I merely moved this logic from WebAudioBufferList but yes, it looks to me like it is to crash on overflow.